lib/sift/value_parser.rb

```diff
@@ -1,11 +1,8 @@
 module Sift
   class ValueParser
-    def initialize(value:, type: nil, options: {})
-      @value = value
-      @supports_boolean = options.fetch(:supports_boolean, false)
-      @supports_ranges = options.fetch(:supports_ranges, false)
-      @supports_json = options.fetch(:supports_json, false)
+    def initialize(value:, type: nil)
```

Comment:
ValueParser can be responsible for determining supports_boolean, supports_ranges, and supports_json from the type, so it doesn't need to be passed in. This helps us parse values for things other than the "chief" parameter.